### PR TITLE
feat: add skeleton code for datadeal module

### PR DESCRIPTION
**Pull request by inchori**
Add skeleton code for `x/datadeal`.

### CODE REVIEW

File: x/datadeal/genesis.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
+	//TODO implement me
+	panic("implement me")
+}
+
+// ExportGenesis returns the capability module's exported genesis.
+func ExportGenesis(ctx sdk.Context, k keeper.Keeper) *types.GenesisState {
+	//TODO implement me
+	panic("implement me")
+}
```
Comment (H4NLee) [inline x/datadeal/genesis.go:20]: I suggest not to panic here. Because chain cannot be initialized with this panic.

**Review by audtlr24 — APPROVED** [review]
lgtm

File: x/datadeal/genesis.go (hunk 1 above)
Comment (inchori) [reply x/datadeal/genesis.go]: Fixed in [351464b](https://github.com/medibloc/panacea-core/pull/520/commits/351464b7b2ff484dd8d194b99a6ec565dba8e3ce).

File: x/datadeal/types/message_deal.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	return nil
+}
+
+func (m *MsgCreateDeal) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *MsgCreateDeal) GetSigners() []sdk.AccAddress {
+	buyerAddress, err := sdk.AccAddressFromBech32(m.BuyerAddress)
```
Comment (gyuguen) [inline x/datadeal/types/message_deal.go:26]: Oh my mistake.
Please change field `buyer_address` to `consumer_address` in `datadea/v2/tx.proto`

**Review by gyuguen — APPROVED** [review]
LGTM
I left an one comment.

File: x/datadeal/types/message_deal.go (hunk 2 above)
Comment (inchori) [reply x/datadeal/types/message_deal.go:26]: In fact, I was curious about this. I'll change it!

File: x/datadeal/types/message_deal.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	bz := ModuleCdc.MustMarshalJSON(m)
+	return sdk.MustSortJSON(bz)
+}
+
+func (m *MsgCreateDeal) GetSigners() []sdk.AccAddress {
+	buyerAddress, err := sdk.AccAddressFromBech32(m.ConsumerAddress)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{buyerAddress}
```
Comment (gyuguen) [inline x/datadeal/types/message_deal.go:30]:
```suggestion
	consumerAddress, err := sdk.AccAddressFromBech32(m.ConsumerAddress)
	if err != nil {
		panic(err)
	}
	return []sdk.AccAddress{consumerAddress}
```

**Pull request merged**